### ISSUE TITLE
GTSAM/python example does not work properly.

### ISSUE THREAD

**Issue by fatrybl** [borglab/gtsam#1675]
## Description

[This example](https://github.com/borglab/gtsam/blob/develop/python/gtsam/examples/OdometryExample.py) in python does not plot marginal ellipsoids for my environment (see below). 
![image](https://github.com/borglab/gtsam/assets/22779281/a57568fa-e3c3-47d7-a61a-05bdf27f1bf9)

## Steps to reproduce

1. try the [example](https://github.com/borglab/gtsam/blob/develop/python/gtsam/examples/OdometryExample.py)

## Environment

Python version: 3.11.5
gtsam: 4.2
numpy: 1.25.2
matplotlib: 3.8.2

**Comment by joehjhuang** [comment]
We have the same issue. It turns out if we edit the source code in `\gtsam\utils\plot.py` in that function `plot_covariance_ellipse_2d` line 155 to:
`e1 = patches.Ellipse(xy=origin,
                         width=np.sqrt(w[0]) * 2 * k,
                         height=np.sqrt(w[1]) * 2 * k,
                         angle=np.rad2deg(angle),
                         fill=False)`
Everything start working again.
Basically give the variables their name and it works.

**Comment by ProfFan** [comment]
@joehjhuang Can you put up a pull request that fixes this? Thank you!

**Comment by varunagrawal** [comment]
This is because Matplotlib's API changed for the `patches.Ellipse` method. I am able to reproduce this with matplotlib 3.7.1. 

The fix is easy, I'll make a PR.
